### ISSUE TITLE
Target ESM over CommonJS

### ISSUE THREAD

**Issue by kasparkallas** [immerjs/use-immer#117]
Currently the module target is set as [CommonJS](https://github.com/immerjs/use-immer/blob/master/tsconfig.json#L8).

Nowadays ESM is a lower common denominator than CommonJS as it's supported in both Node and browser runtimes.

I have a pure ESM library which I serve unbundled (only tsc output) and the usage of the use-immer's `require` causes some trouble.

**Comment by mweststrate** [comment]
PR welcome

**Comment by irishcarbomb777** [comment]
I just ran into this issue also. I solved it by setting {"type": "module"} in the package.json for node_modules/use-immer (In other words for anyone else with the issue, add that line to the package.json of the installed use-immer package within the node_modules folder). I've never had an issue with the package until building a react-based 'ink' cli tool for node.

**Comment by Seize-cf** [comment]
It appears that `microbundle` may not have a dependency on `tsconfig.json`. Perhaps changing the `module` to `ESNext` could potentially solve the issue.

**Comment by k1sul1** [comment]
> PR welcome

The way I see it, there's 2 PRs open regarding this. I don't know or care if they work to solve the problem, but I see that they don't have any comments or any signs life really. 

I will not be making a 3rd just to also get ignored.

> I just ran into this issue also. I solved it by setting {"type": "module"} in the package.json for node_modules/use-immer (In other words for anyone else with the issue, add that line to the package.json of the installed use-immer package within the node_modules folder). I've never had an issue with the package until building a react-based 'ink' cli tool for node.

Thanks for this. Simply adding `"module": true` to the package.json in `node_modules/use-immer` and then rebuilding did the trick.

Went from this error

```
import { useImmerReducer } from "use-immer";
         ^^^^^^^^^^^^^^^
SyntaxError: Named export 'useImmerReducer' not found. The requested module 'use-immer' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'use-immer';
const { useImmerReducer } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at ModuleJob.run (node:internal/modules/esm/module_job:190:5)
``` 

to no error in 2 minutes. 

I'm using this on top of a Remix 2.8 project, FWIW. We *tried* to migrate to ESM almost six months ago. We failed, because of this package. It was the only dependency that did not play well with ESM. Now we're in a situation where we can't keep using CJS as we're migrating to Vite

I have fixed the problem for now using `patch-package`, but I guess I'm off to finding another solution, since useImmer is unusable in 2024, and displays no signs of that changing.

**Comment by mweststrate** [comment]
Hopefully fixed with 0.10.0, see #122 by @fedemp. Give it a try (usually ESM conversions break somewhere in my experience :-P)
